pkg/clustersync/clusterSync.go

```diff
@@ -4,24 +4,252 @@ package clustersync
 
 import (
 	"context"
+	"encoding/json"
+	"strings"
+	"sync"
+	"time"
 
+	clusterv1beta1 "github.com/stolostron/multicloud-operators-foundation/pkg/apis/internal.open-cluster-management.io/v1beta1"
 	"github.com/stolostron/search-indexer/pkg/config"
+	"github.com/stolostron/search-indexer/pkg/database"
+	"github.com/stolostron/search-indexer/pkg/model"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
 	klog "k8s.io/klog/v2"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
 )
 
+var dynamicClient dynamic.Interface
+var dao database.DAO
+var client *kubernetes.Clientset
+var mux sync.Mutex
+
+const managedClusterGVR = "managedclusters.v1.cluster.open-cluster-management.io"
+const managedClusterInfoGVR = "managedclusterinfos.v1beta1.internal.open-cluster-management.io"
+const lockName = "search-indexer.open-cluster-management.io"
+
 func ElectLeaderAndStart(ctx context.Context) {
-	client := config.Cfg.KubeClient
-	lockName := "search-indexer.open-cluster-management.io"
+	client = config.Cfg.KubeClient
 	podName := config.Cfg.PodName
 	podNamespace := config.Cfg.PodNamespace
-
+	if (database.DAO{} == dao) {
+		dao = database.NewDAO(nil)
+	}
 	lock := getNewLock(client, lockName, podName, podNamespace)
 	runLeaderElection(ctx, lock)
 }
 
-func syncClusters(ctx context.Context) {
-	klog.Info("TODO: Start Sync clusters here.")
+// Watches ManagedCluster objects and updates the database with a Cluster node.
+func watchClusters(ctx context.Context) {
```

Comment:
Could you rename to `syncClusters()`? There's several places where we should change `watch` to `sync`.

Sorry I changed this, I thought `sync` is more accurate.